File: src/main/java/com/droiddevgeeks/crashsink/CrashAttributor.java
package com.droiddevgeeks.crashsink;

/** Decides whether a crash originated in owned code (top application frame). */
public final class CrashAttributor {

    private final String ownedPrefix;

    public CrashAttributor(final String ownedPrefix) {
        this.ownedPrefix = ownedPrefix;
    }

    public boolean isOurs(final Throwable throwable) {
        if (throwable == null) {
            return false;
        }
        final StackTraceElement origin = CrashFrames.firstApplicationFrame(
                CrashFrames.deepestCause(throwable).getStackTrace());
        return origin != null && CrashFrames.isOurs(origin.getClassName(), ownedPrefix);
    }
}

File: crashsink/src/main/kotlin/com/droiddevgeeks/crashsink/CrashAttributor.kt
package com.droiddevgeeks.crashsink

/** Decides whether a crash originated in owned code (top application frame). */
class CrashAttributor(val ownedPrefix: String) {

    fun isOurs(throwable: Throwable?): Boolean {
        if (throwable == null) {
            return false
        }
        val origin = CrashFrames.firstApplicationFrame(
            CrashFrames.deepestCause(throwable).stackTrace
        )
        return origin != null && CrashFrames.isOurs(origin.className, ownedPrefix)
    }
}
